src/cascadia/TerminalSettingsModel/HashUtils.h

```diff
@@ -17,54 +17,37 @@ Revision History:
 
 #pragma once
 
-namespace Microsoft::Terminal::Settings::Model::HashUtils
-{
-    // This is a helper template function for hashing multiple variables in conjunction to each other.
-    template<typename T>
-    constexpr size_t HashProperty(const T& val)
-    {
-        std::hash<T> hashFunc;
-        return hashFunc(val);
-    }
-
-    template<typename T, typename... Args>
-    constexpr size_t HashProperty(const T& val, Args&&... more)
-    {
-        // Inspired by boost::hash_combine, which causes this effect...
-        //   seed ^= hash_value(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
-        // Source: https://www.boost.org/doc/libs/1_35_0/doc/html/boost/hash_combine_id241013.html
-        const auto seed{ HashProperty(val) };
-        return seed ^ (0x9e3779b9 + (seed << 6) + (seed >> 2) + HashProperty(std::forward<Args>(more)...));
-    }
+#include <til/hash.h>
 
-    template<>
-    constexpr size_t HashProperty(const til::color& val)
-    {
-        return HashProperty(val.a, val.r, val.g, val.b);
-    }
-
-#ifdef WINRT_Windows_Foundation_H
+namespace til
+{
     template<typename T>
-    constexpr size_t HashProperty(const winrt::Windows::Foundation::IReference<T>& val)
+    struct hash_trait<winrt::Windows::Foundation::IReference<T>>
     {
-        return val ? HashProperty(val.Value()) : 0;
-    }
-#endif
+        constexpr void operator()(hasher& h, const winrt::Windows::Foundation::IReference<T>& v) const noexcept
+        {
+            if (v)
+            {
+                h.write(v.Value());
+            }
+        }
+    };
 
-#ifdef WINRT_Windows_UI_H
     template<>
-    constexpr size_t HashProperty(const winrt::Windows::UI::Color& val)
+    struct hash_trait<winrt::Microsoft::Terminal::Settings::Model::NewTerminalArgs>
     {
-        return HashProperty(til::color{ val });
-    }
-#endif
+        void operator()(hasher& h, const winrt::Microsoft::Terminal::Settings::Model::NewTerminalArgs& value) const noexcept
+        {
+            h.write(winrt::get_abi(value));

```

Comment:
wait a minute, wat? this hashes the pointer to the `NewTerminalArgs`, doesn't it? That should never compare-same...

Reply:
Yep and that's exactly what WinRT's std::hash implementation does for any WinRT pointer.

Reply:
In that case, I don't understand how this is supposed to work-

https://github.com/microsoft/terminal/pull/11887/files#diff-495c1454343b5dab4b655ba0ca18b95ec179907feb371e1e0ff494cf7a7bb21eR369

We used to write the output of `TerminalArgs().Hash()` (using the `HashProperty()` indirector) to the Action's hash, and now we're writing `TerminalArgs()` itself -- so we get the pointer out of it.